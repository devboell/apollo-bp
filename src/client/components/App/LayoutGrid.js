import styled from 'styled-components'

const LayoutGrid = styled.div`
  display: grid;
  grid-gap: 1em;
  grid-template-columns: 1fr 10fr;
  grid-template-areas:
     "homeIcon header"
     "artistNav artistNav"
     "content content"
`

export default LayoutGrid
