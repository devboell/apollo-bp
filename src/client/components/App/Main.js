import styled from 'styled-components'

const Main = styled.main`
  display: flex;
  justify-content: space-around;
`

export default Main
