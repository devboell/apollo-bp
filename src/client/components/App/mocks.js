
import ARTISTS_QUERY from 'graphql/ArtistsQuery'
import artistsFixt from 'fixtures/artistsQuery'

import ARTIST_BY_PATH_QUERY from 'graphql/ArtistByPathQuery'
import artistByPathFixt from 'fixtures/artistByPathQuery'

export const artists = {
  request: { query: ARTISTS_QUERY },
  result: { data: { artists: artistsFixt } },
}

export const artistByPath = {
  request: { query: ARTIST_BY_PATH_QUERY, variables: { path: 'heroshige' } },
  result: { data: { artistByPath: artistByPathFixt } },
}

export const mocks = [
  artists,
  artistByPath,
]
