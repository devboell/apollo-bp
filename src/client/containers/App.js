import React from 'react'

import artists from 'data/artists.json'
import List from 'components/List'

const App = () =>
  <List {...{ artists }} />

export default App
