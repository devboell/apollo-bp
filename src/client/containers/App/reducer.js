import artists from 'data/artists.json'

const initialState = {
  artists,
}

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state
  }
}
