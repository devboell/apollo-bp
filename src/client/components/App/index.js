import React from 'react'

import HomeIconArea from 'components/HomeIconArea'
import HeaderArea from 'components/HeaderArea'
import ArtistNavArea from 'components/ArtistNavArea'
import ContentArea from 'components/ContentArea'

import LayoutGrid from './LayoutGrid'

const App = () => (
  <LayoutGrid>
    <HomeIconArea />

  </LayoutGrid>
)
/*
    <HeaderArea />
    <ArtistNavArea />
    <ContentArea />
*/

export default App
