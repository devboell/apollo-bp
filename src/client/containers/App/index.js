import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import ArtistsNav from 'components/ArtistsNav'
import Header from 'components/Header'
import Content from 'components/Content'

import Main from './Main'

const App = ({ artists }) =>
  <div>
    <Header />
    <Main>
      <ArtistsNav {...{ artists }} />
      <Content />
    </Main>
  </div>


App.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string })).isRequired,
}

const mapStateToProps = state => ({
  artists: state.app.artists,
})


export default withRouter(connect(mapStateToProps)(App))
