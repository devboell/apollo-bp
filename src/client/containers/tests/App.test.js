import React from 'react'
import renderer from 'react-test-renderer'
import App from '../App'

it('App component snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
});
