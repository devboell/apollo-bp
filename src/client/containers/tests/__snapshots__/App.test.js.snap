// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
<ul
  className="ihkFxl"
>
  <li
    className="bfXghp"
  >
    <h3>
      Heroshige
    </h3>
  </li>
  <li
    className="bfXghp"
  >
    <h3>
      Hokusai
    </h3>
  </li>
  <li
    className="bfXghp"
  >
    <h3>
      Utamaro
    </h3>
  </li>
  <li
    className="bfXghp"
  >
    <h3>
      Kunisada
    </h3>
  </li>
</ul>
`;
